src/vtex-ng-pagination.js

```diff
@@ -45,25 +47,22 @@ angular.module('vtex.ngPagination', []).directive('vtPagination', [
         })
 
         scope.$watch('currentPage', function(newValue) {
-          scope.currentPageDisplay = newValue
+          scope.currentPageDisplay = Math.min(newValue, scope.maxPage)
           scope.verifyBtns(scope.currentPageDisplay)
         })
 
         // Methods on scope
         scope.setCurrentPage = function(newValue) {
-          if (!isNaN(newValue)) {
-            if (newValue && newValue <= 0) {
-              scope.currentPage = 1
-            } else if (newValue && newValue > scope.pageCount) {
-              scope.currentPage = scope.pageCount ? scope.pageCount : 1
-            } else if (newValue != scope.currentPage) {
-              scope.currentPage = newValue
-            }
+          if (newValue && newValue <= 0) {
+            scope.currentPage = 1
+          } else {
+            scope.currentPage = Math.min(newValue, scope.maxPage)
```

Comment:
If newValue is null or undefined, this may cause unexpected behavior. It was being treated before, I don't know if it should now. Just fyi

Reply:
he did set it with `|| 1` in the beginning of the file 😛